libs/gi/ui/src/components/GOAdWrapper.tsx

```diff
@@ -0,0 +1,54 @@
+import type { AdDims, AdProps } from '@genshin-optimizer/common/ad'
+import {
+  getGOAd,
+  getGODrakeAd,
+  getGOLootbarAd,
+  getZOBannerAd,
+  getZORectAd,
+} from '@genshin-optimizer/common/ad'
+import { CardThemed } from '@genshin-optimizer/common/ui'
+import { getRandomElementFromArray } from '@genshin-optimizer/common/util'
+import { notEmpty } from '@genshin-optimizer/common/util'
+import { useMemo } from 'react'
+import type { FunctionComponent, ReactNode } from 'react'
+/**
+ * A component that aggregates all ads shown on GO
+ */
+export function GOAdWrapper({ sx = {}, bgt = 'light', children }: AdProps) {
```

Comment:
can maybe use SxProps type from mui/material or something like that?

Reply:
yes https://github.com/frzyc/genshin-optimizer/blob/ad_lib/libs/common/ad/src/type.tsx#L9

Reply:
oh, then why is there 'as any' below?